homeassistant/components/rainbird/switch.py

```diff
@@ -80,6 +123,12 @@ def turn_off(self, **kwargs):
         if response and response["type"] == "AcknowledgeResponse":
             self._state = False
 
+    def start_irrigation(self, duration: int):
+        """Turn the irrigation on."""
+        response = self._rainbird.startIrrigation(int(self._zone), int(duration))
+        if response and response["type"] == "AcknowledgeResponse":
```

Comment:
Also this is not true already. In pyrainbird there are returned True or False. REbase on my pullrequest #26393 where changes are made.